pkg/apis/application/v1alpha1/repository_types.go

```diff
@@ -116,6 +119,9 @@ type Repository struct {
 	InsecureOCIForceHttp bool `json:"insecureOCIForceHttp,omitempty" protobuf:"bytes,26,opt,name=insecureOCIForceHttp"` //nolint:revive //FIXME(var-naming)
 	// Depth specifies the depth for shallow clones. A value of 0 or omitting the field indicates a full clone.
 	Depth int64 `json:"depth,omitempty" protobuf:"bytes,27,opt,name=depth"`
+	// UserAgent specifies a custom User-Agent string for HTTP requests to this Helm repository.
+	// If not set, a default User-Agent will be used.
+	UserAgent string `json:"userAgent,omitempty" protobuf:"bytes,28,opt,name=userAgent"`
```

Comment:
I think this should be set as an environment variable on the repo-server rather than being set on the repository.

Reply:
> Would you prefer a single global User-Agent for all Helm repos, or should there be a way to override per-repository?

IMO this should be a global option. Up until now there has been no need for setting a header for a Helm repository, and I don't anticipate there will be much of a need to configure this on a per repository basis either.

>Are there specific concerns with the current per-repository approach?

We already have too many options on the Repository object as-is.

> Would a hybrid approach (env var as default + per-repo override) make sense?

No